planner/core/exhaust_physical_plans.go

```diff
@@ -427,7 +427,11 @@ func (p *LogicalJoin) getIndexJoinByOuterIdx(prop *property.PhysicalProperty, ou
 			keyOff2IdxOff[i] = 0
 		}
 		if pkMatched {
-			innerPlan := p.constructInnerTableScan(ds, pkCol, outerJoinKeys, us)
+			innerPlan, err := p.constructInnerTableScan(ds, pkCol, outerJoinKeys, us)
+			if err != nil {
+				logutil.BgLogger().Error("construct inner table scan error", zap.Error(err))
```

Comment:
How about setting this log level to `DEBUG`? Users don't need to be awarded about why the index join can't be chosen. This `ERROR` log may make them confusion.

Reply:
If this error is not null, it represents that our implicit assumption is broken, which should never happen.
This is a severe error, so it's better to use the `ERROR` level?

Reply:
Actually, I think using `assert(err, IsNil)` is more appropriate, but Golang has no assertion.

Reply:
PTAL

Reply:
I think we should panic if reach the unreachable path.

Reply:
It's better to return the error rather than just panic. It helps us to:
1. continue to server new request from the users
2. discover the internal error and make improvement.

Reply:
This method `getIndexJoinByOuterIdx` is the outermost layer to handle an error when building plans 🤣.
If there is an error in this method, it logs the error and returns `nil` to let the planner do not choose `IndexJoin`.

Reply:
OK